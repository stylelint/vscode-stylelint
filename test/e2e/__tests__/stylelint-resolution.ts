import { openDocument, waitForDiagnostics, assertDiagnostics, closeAllEditors } from '../helpers';

describe('Stylelint resolution', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should resolve Stylelint using local node_modules', async () => {
		const document = await openDocument('defaults/local-stylelint/test.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'fake',
				message: 'Fake result from resolve-local',
				range: [0, 0, 0, 1],
				severity: 'error',
			},
		]);
	});

	describe('when using "stylelint.stylelintPath"', () => {
		it('should resolve Stylelint', async () => {
			const document = await openDocument('stylelint-path/test.css');
			const diagnostics = await waitForDiagnostics(document);

			assertDiagnostics(diagnostics, [
				{
					code: 'fake',
					message: 'Fake result',
					range: [0, 0, 0, 1],
					severity: 'error',
				},
			]);
		});
	});

	it('should resolve Stylelint using PnP', async () => {
		const document = await openDocument('defaults/yarn-pnp/test.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'fake',
				message: 'Fake result from yarn-pnp',
				range: [0, 0, 0, 1],
				severity: 'error',
			},
		]);
	});

	it('should resolve Stylelint using Yarn 2.x PnP', async () => {
		const document = await openDocument('defaults/yarn-2-pnp/test.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'fake',
				message: 'Fake result from yarn-2-pnp',
				range: [0, 0, 0, 1],
				severity: 'error',
			},
		]);
	});
});
