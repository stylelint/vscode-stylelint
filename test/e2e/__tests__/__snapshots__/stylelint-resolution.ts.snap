// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stylelint resolution should resolve Stylelint using "stylelint.stylelintPath" 1`] = `
[
  {
    "code": "fake",
    "message": "Fake result",
    "range": {
      "end": {
        "character": 1,
        "line": 0,
      },
      "start": {
        "character": 0,
        "line": 0,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;

exports[`Stylelint resolution should resolve Stylelint using PnP 1`] = `
[
  {
    "code": "fake",
    "message": "Fake result from yarn-pnp",
    "range": {
      "end": {
        "character": 1,
        "line": 0,
      },
      "start": {
        "character": 0,
        "line": 0,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;

exports[`Stylelint resolution should resolve Stylelint using Yarn 2.x PnP 1`] = `
[
  {
    "code": "fake",
    "message": "Fake result from yarn-2-pnp",
    "range": {
      "end": {
        "character": 1,
        "line": 0,
      },
      "start": {
        "character": 0,
        "line": 0,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;

exports[`Stylelint resolution should resolve Stylelint using local node_modules 1`] = `
[
  {
    "code": "fake",
    "message": "Fake result from resolve-local",
    "range": {
      "end": {
        "character": 1,
        "line": 0,
      },
      "start": {
        "character": 0,
        "line": 0,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
